tools/reviewer/job/impl/SubmitterTask.java

```diff
@@ -0,0 +1,187 @@
+                      this.firestoreClient.getDocumentFromCollection(
+                          String.format(CI_RESPONSES_HISTORY_PATH, diff.getId()),
+                          CIResponse.newBuilder());
+                  CIResponse response = (CIResponse) responseWithId.message();
+                  return Pair.of(diff, response);
+                })
+            // only proceed if CIResponse was present
+            .filter(diffCiPair -> diffCiPair.getValue() != null)
+            // only proceed if TargetResult.success = true for all Results
+            .filter(
+                diffCiPair ->
+                    diffCiPair
+                        .getValue()
+                        .getResultsList()
+                        .stream()
+                        .allMatch(CIResponse.TargetResult::getSuccess))
+            .forEach(
+                diffCiPair -> {
+                  Diff diff = diffCiPair.getKey();
+                  CIResponse response = diffCiPair.getValue();
+
+                  boolean shouldPush = true;
+                  List<GitRepo> gitRepos = new ArrayList<>();
+
+                  CIRequest.Builder newRequestBuilder =
+                      CIRequest.newBuilder().setForSubmission(true);
+
+                  for (CIResponse.TargetResult targetResult : response.getResultsList()) {
+                    CIRequest.Target target = targetResult.getTarget();
+
+                    Repo repo = target.getRepo();
+                    String repoPath =
+                        fileUtils.joinPaths(
+                            fileUtils.getCurrentWorkingDirectory(), "submitter", repo.getId());
+                    GitRepo gitRepo = this.gitRepoFactory.create(repoPath);
+                    gitRepos.add(gitRepo);
+
+                    try {
+                      if (fileUtils.folderEmptyOrNotExists(repoPath)) {
+                        gitRepo.cloneRepo(repo.getUrl(), repoPath);
+                      } else {
+                        gitRepo.pull();
+                      }
+                    } catch (IOException e) {
+                      e.printStackTrace();
+                    }
+
+                    String branch = String.format("origin/D%d", diff.getId());
+                    gitRepo.switchBranch(branch);
+
+                    newRequestBuilder =
+                        newRequestBuilder.addTarget(
+                            CIRequest.Target.newBuilder()
+                                .setRepo(repo)
+                                .setCommitId(gitRepo.getHeadCommitId())
+                                .build());
+
+                    if (gitRepo.getHeadCommitId().equals(target.getCommitId())) {
+                      // latest result for CI matches last commit on the branch
+
+                      gitRepo.switchBranch("master");
+                      boolean mergingResult = gitRepo.merge("D%d", true);
+                      if (!mergingResult || gitRepo.hasChanges("master")) {
```

Comment:
Do we have a place to store the failed result? For example - `Merge failed` or `Commit out of date`.

Reply:
@oferb Nope. what is the best way to handle it?

Reply:
@oferb do you mean we should add `field` to proto or new proto message class?

Reply:
I don't mind. Could be a message, or 2 separate fields in an existing message.

Reply:
@oferb I think it makes sense to have `SubmitterMergeResult` as new proto message which will be stored in `/reviewer/submitter/<diffId>/history/`. I'll create a message & add a `TODO` to store result